Stripe/Payments/STPPaymentHandler.m

```diff
@@ -62,23 +62,39 @@ - (void)confirmPayment:(STPPaymentIntentParams *)paymentParams
     }
     self.inProgress = YES;
     __weak __typeof(self) weakSelf = self;
+    // wrappedCompletion ensures we perform some final logic before calling the completion block.
+    STPPaymentHandlerActionPaymentIntentCompletionBlock wrappedCompletion = ^(STPPaymentHandlerActionStatus status, STPPaymentIntent *paymentIntent, NSError *error) {
+        // Reset our internal state
+        weakSelf.inProgress = NO;
+        // Ensure the .succeeded case returns a PaymentIntent in the expected state.
+        if (status == STPPaymentHandlerActionStatusSucceeded) {
+            if (error == nil && paymentIntent != nil && (paymentIntent.status == STPPaymentIntentStatusSucceeded || paymentIntent.status == STPPaymentIntentStatusRequiresCapture)) {
+                completion(STPPaymentHandlerActionStatusSucceeded, paymentIntent, nil);
+            } else {
+                NSAssert(NO, @"Calling completion with invalid state");
+                completion(STPPaymentHandlerActionStatusFailed, paymentIntent, error ?: [self _errorForCode:STPPaymentHandlerIntentStatusErrorCode userInfo:nil]);
```

Comment:
a `self` snuck in here.
We should add a `strongSelf` here so that once the block starts executing we know it will finish

Reply:
gah!

Reply:
Scary bug :( thanks!